### DISCUSSION TITLE
PlayWaveサンプルでのHW改造について

### DISCUSSION THREAD

**Discussion by y-ohe** [algyan/XIAOGYAN#44] [Ideas]
@matsujirushi さん
CRのローパスフィルタ回路を構成するためにR7=1K、C6＝0.01uにする改造を載せていますが、1Kの抵抗も0.01uのコンデンサも元の部品表(BOM)にありません。
R7=100、C6=0.1uで改造すれば、同一特性で100Ωの抵抗を追加するのみで、C1で使う予定の0.1uの余剰部品を有効活用できると思います。
いかかでしょうか？
追加となる100ΩはLEDの制限抵抗用(R5, R6, R12)としても流用もできそうなので、部品追加しても大丈夫な気がしています。

**Comment by matsujirushi** [comment]
@y-ohe さん、アドバイスありがとうございます！

Rを100Ωにすると、XIAOのGPIOからの電流が不足するような気がしますが、このあたり大丈夫でしょうか？

**Reply by y-ohe** [reply]
GPIOの出力電流不足については、LPFの過渡での話なので、計算も難しいですね。
DC特性を確認しましたが、一応IOH=40mA(typ)、IOL=28ｍA(typ)ですし、3.3V÷100Ω＝33ｍAなので、ギリギリなのかなと。
平均電流で見れば、PWM出力の周波数次第にもなりますが、最大でもこの50％ぐらいになるかと考えます。

実際に過渡応答時に電流が不足する場合は、LPFのコンデンサの充放電が、理論通りにできずLPF特性が悪化することと、計算上でギリギリなので、ちょっとデバイスを酷使することになるかも、といった感じですね。
充放電にかかる時間も計算上だいたい0.1msec程度のようなので、10kHzあたりの周波数でPWM出力をすると最大電流での充放電がされる状況になると思われます。

100Ω、0.1uFでのLPFでは、GPIO出力電流が気になる場合は、10kΩ、102(0.001uF)にするのもありかと。
1608サイズの102は、秋月でも普通に入手可能なようです。

**Reply by matsujirushi** [reply]
Note
#45 で対応中。

**Reply by matsujirushi** [reply]
#45 で対応完了。
